src/west/manifest.py

```diff
@@ -217,6 +239,26 @@ class MalformedManifest(Exception):
 
 # Definitions for Manifest attribute types.
 
+class WestMeta:
+
+    __slots__ = 'url revision'.split()
```

Comment:
`__slots__` is a bit overkill here by the way. It makes a difference in Kconfiglib, because parsing in Python is slow, but this code isn't very performance-sensitive. :)

Reply:
Same, I took this from your code in `manifest.py`. Well, I assume it's your code, it might be @mbolivar's.

Reply:
Most of that code got moved from `project.py`. There weren't any classes when it was in there.

Reply:
Yup, that's mine.